src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -160,6 +161,30 @@ public async Task AcknowledgeBatchProcessed(long subscriptionId, string name, st
                 DocumentsToResend = docsToResend
             };
 
+            var state = _serverStore.Engine.CurrentState;
+            if (command.ChangeVector == nameof(Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange) &&
```

Comment:
why is it here and not under `SendNoopAck` method?

Reply:
Added a separate method

Reply:
why not go through this one:
https://github.com/ravendb/ravendb/blob/4faface9207731780ab27fa4dbbb0e399443a563/src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionsState.cs#L646C25-L646C26

Reply:
done